yt/fields/astro_fields.py

```diff
@@ -124,3 +125,48 @@ def _entropy(field, data):
     registry.add_field(
         (ftype, "entropy"), sampling_type="local", units="keV*cm**2", function=_entropy
     )
+
+    def _lorentz_factor(field, data):
+        b2 = data[ftype, "velocity_magnitude"].to_value("c")
+        b2 *= b2
+        return 1.0 / np.sqrt(1.0 - b2)
```

Comment:
Hi @zengbs, 

This particular field definition occurs when we do not have the four-velocities available--i.e., when we only have 3-velocity. It's a general definition for all codes in yt, and will probably suffice for most non-relativistic situations. Relativistic codes (like GAMER) will override it (see [here](https://github.com/yt-project/yt/pull/3033/files#diff-28ba9a9751b3bd2b614b7d004fcef3bea29d3f5651ca0c9ac5e1d5c025af8cd5R130)) to avoid the cancellation.

Reply:
@jzuhone Understand. Thanks!